### ISSUE TITLE
fabric8:build failing on openshift 

### ISSUE THREAD

**Issue by rohanKanojia** [fabric8io/fabric8-maven-plugin#1704]
### Description
Seems like recent kubernetes client upgrade to 4.4.2 has broken fabric8:build goal on Openshift. Right now it seems to be breaking with this error:
```
~/work/repos/fabric8-maven-plugin/samples/spring-boot : $ mvn fabric8:build
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------< io.fabric8:fabric8-maven-sample-spring-boot >-------------
[INFO] Building Fabric8 Maven :: Sample :: Spring Boot Web 4.3-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- fabric8-maven-plugin:4.3-SNAPSHOT:build (default-cli) @ fabric8-maven-sample-spring-boot ---
[INFO] F8: Running in OpenShift mode
[INFO] F8: Using OpenShift build with strategy S2I
[INFO] F8: Running generator spring-boot
[INFO] F8: spring-boot: Using Container image fabric8/s2i-java:2.3 as base / builder
[INFO] Copying files to /home/Rohaan/work/repos/fabric8-maven-plugin/samples/spring-boot/target/docker/fabric8/fabric8-maven-sample-spring-boot/latest/build/maven
[INFO] Building tar: /home/Rohaan/work/repos/fabric8-maven-plugin/samples/spring-boot/target/docker/fabric8/fabric8-maven-sample-spring-boot/latest/tmp/docker-build.tar
[INFO] F8: [fabric8/fabric8-maven-sample-spring-boot:latest] "spring-boot": Created docker source tar /home/Rohaan/work/repos/fabric8-maven-plugin/samples/spring-boot/target/docker/fabric8/fabric8-maven-sample-spring-boot/latest/tmp/docker-build.tar
[INFO] F8: Updating BuildServiceConfig fabric8-maven-sample-spring-boot-s2i for Source strategy
[INFO] F8: Adding to ImageStream fabric8-maven-sample-spring-boot
[INFO] F8: Starting Build fabric8-maven-sample-spring-boot-s2i
[WARNING] Exec Failure
java.io.InterruptedIOException: executor rejected
    at okhttp3.RealCall$AsyncCall.executeOn (RealCall.java:185)
    at okhttp3.Dispatcher.promoteAndExecute (Dispatcher.java:186)
    at okhttp3.Dispatcher.enqueue (Dispatcher.java:137)
    at okhttp3.RealCall.enqueue (RealCall.java:126)
    at okhttp3.internal.ws.RealWebSocket.connect (RealWebSocket.java:193)
    at okhttp3.OkHttpClient.newWebSocket (OkHttpClient.java:435)
    at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager.runWatch (WatchConnectionManager.java:158)
    at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager.<init> (WatchConnectionManager.java:100)
    at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager.<init> (WatchConnectionManager.java:105)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.watch (BaseOperation.java:734)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.watch (BaseOperation.java:720)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.watch (BaseOperation.java:63)
    at io.fabric8.maven.core.service.openshift.OpenshiftBuildService.waitUntilPodIsReady (OpenshiftBuildService.java:575)
    at io.fabric8.maven.core.service.openshift.OpenshiftBuildService.waitForOpenShiftBuildToComplete (OpenshiftBuildService.java:528)
    at io.fabric8.maven.core.service.openshift.OpenshiftBuildService.build (OpenshiftBuildService.java:130)
    at io.fabric8.maven.plugin.mojo.build.BuildMojo.buildAndTag (BuildMojo.java:296)
    at io.fabric8.maven.docker.BuildMojo.processImageConfig (BuildMojo.java:105)
    at io.fabric8.maven.docker.BuildMojo.executeInternal (BuildMojo.java:61)
    at io.fabric8.maven.plugin.mojo.build.BuildMojo.executeInternal (BuildMojo.java:261)
    at io.fabric8.maven.docker.AbstractDockerMojo.execute (AbstractDockerMojo.java:230)
    at io.fabric8.maven.plugin.mojo.build.BuildMojo.execute (BuildMojo.java:211)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.util.concurrent.RejectedExecutionException: Task okhttp3.RealCall$AsyncCall@6f5e16cf rejected from java.util.concurrent.ThreadPoolExecutor@750a04ec[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]
    at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution (ThreadPoolExecutor.java:2063)
    at java.util.concurrent.ThreadPoolExecutor.reject (ThreadPoolExecutor.java:830)
    at java.util.concurrent.ThreadPoolExecutor.execute (ThreadPoolExecutor.java:1379)
    at okhttp3.RealCall$AsyncCall.executeOn (RealCall.java:182)
    at okhttp3.Dispatcher.promoteAndExecute (Dispatcher.java:186)
    at okhttp3.Dispatcher.enqueue (Dispatcher.java:137)
    at okhttp3.RealCall.enqueue (RealCall.java:126)
    at okhttp3.internal.ws.RealWebSocket.connect (RealWebSocket.java:193)
    at okhttp3.OkHttpClient.newWebSocket (OkHttpClient.java:435)
    at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager.runWatch (WatchConnectionManager.java:158)
    at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager.<init> (WatchConnectionManager.java:100)
    at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager.<init> (WatchConnectionManager.java:105)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.watch (BaseOperation.java:734)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.watch (BaseOperation.java:720)
    at io.fabric8.kubernetes.client.dsl.base.BaseOperation.watch (BaseOperation.java:63)
    at io.fabric8.maven.core.service.openshift.OpenshiftBuildService.waitUntilPodIsReady (OpenshiftBuildService.java:575)
    at io.fabric8.maven.core.service.openshift.OpenshiftBuildService.waitForOpenShiftBuildToComplete (OpenshiftBuildService.java:528)
    at io.fabric8.maven.core.service.openshift.OpenshiftBuildService.build (OpenshiftBuildService.java:130)
    at io.fabric8.maven.plugin.mojo.build.BuildMojo.buildAndTag (BuildMojo.java:296)
    at io.fabric8.maven.docker.BuildMojo.processImageConfig (BuildMojo.java:105)
    at io.fabric8.maven.docker.BuildMojo.executeInternal (BuildMojo.java:61)
    at io.fabric8.maven.plugin.mojo.build.BuildMojo.executeInternal (BuildMojo.java:261)
    at io.fabric8.maven.docker.AbstractDockerMojo.execute (AbstractDockerMojo.java:230)
    at io.fabric8.maven.plugin.mojo.build.BuildMojo.execute (BuildMojo.java:211)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
[ERROR] F8: Could not watch pod: io.fabric8.kubernetes.client.KubernetesClientException: Failed to start websocket
[INFO] F8: Waiting for build fabric8-maven-sample-spring-boot-s2i-10 to complete...
[ERROR] Log Callback Failure.
java.io.InterruptedIOException: executor rejected
    at okhttp3.RealCall$AsyncCall.executeOn (RealCall.java:185)
    at okhttp3.Dispatcher.promoteAndExecute (Dispatcher.java:186)
    at okhttp3.Dispatcher.enqueue (Dispatcher.java:137)
    at okhttp3.RealCall.enqueue (RealCall.java:126)
    at io.fabric8.kubernetes.client.dsl.internal.PodOperationsImpl.watchLog (PodOperationsImpl.java:207)
    at io.fabric8.kubernetes.client.dsl.internal.PodOperationsImpl.watchLog (PodOperationsImpl.java:197)
    at io.fabric8.kubernetes.client.dsl.internal.PodOperationsImpl.watchLog (PodOperationsImpl.java:68)
    at io.fabric8.maven.core.service.openshift.OpenshiftBuildService.waitForOpenShiftBuildToComplete (OpenshiftBuildService.java:530)
    at io.fabric8.maven.core.service.openshift.OpenshiftBuildService.build (OpenshiftBuildService.java:130)
    at io.fabric8.maven.plugin.mojo.build.BuildMojo.buildAndTag (BuildMojo.java:296)
    at io.fabric8.maven.docker.BuildMojo.processImageConfig (BuildMojo.java:105)
    at io.fabric8.maven.docker.BuildMojo.executeInternal (BuildMojo.java:61)
    at io.fabric8.maven.plugin.mojo.build.BuildMojo.executeInternal (BuildMojo.java:261)
    at io.fabric8.maven.docker.AbstractDockerMojo.execute (AbstractDockerMojo.java:230)
    at io.fabric8.maven.plugin.mojo.build.BuildMojo.execute (BuildMojo.java:211)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.util.concurrent.RejectedExecutionException: Task okhttp3.RealCall$AsyncCall@fe09383 rejected from java.util.concurrent.ThreadPoolExecutor@750a04ec[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]
    at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution (ThreadPoolExecutor.java:2063)
    at java.util.concurrent.ThreadPoolExecutor.reject (ThreadPoolExecutor.java:830)
    at java.util.concurrent.ThreadPoolExecutor.execute (ThreadPoolExecutor.java:1379)
    at okhttp3.RealCall$AsyncCall.executeOn (RealCall.java:182)
    at okhttp3.Dispatcher.promoteAndExecute (Dispatcher.java:186)
    at okhttp3.Dispatcher.enqueue (Dispatcher.java:137)
    at okhttp3.RealCall.enqueue (RealCall.java:126)
    at io.fabric8.kubernetes.client.dsl.internal.PodOperationsImpl.watchLog (PodOperationsImpl.java:207)
    at io.fabric8.kubernetes.client.dsl.internal.PodOperationsImpl.watchLog (PodOperationsImpl.java:197)
    at io.fabric8.kubernetes.client.dsl.internal.PodOperationsImpl.watchLog (PodOperationsImpl.java:68)
    at io.fabric8.maven.core.service.openshift.OpenshiftBuildService.waitForOpenShiftBuildToComplete (OpenshiftBuildService.java:530)
    at io.fabric8.maven.core.service.openshift.OpenshiftBuildService.build (OpenshiftBuildService.java:130)
    at io.fabric8.maven.plugin.mojo.build.BuildMojo.buildAndTag (BuildMojo.java:296)
    at io.fabric8.maven.docker.BuildMojo.processImageConfig (BuildMojo.java:105)
    at io.fabric8.maven.docker.BuildMojo.executeInternal (BuildMojo.java:61)
    at io.fabric8.maven.plugin.mojo.build.BuildMojo.executeInternal (BuildMojo.java:261)
    at io.fabric8.maven.docker.AbstractDockerMojo.execute (AbstractDockerMojo.java:230)
    at io.fabric8.maven.plugin.mojo.build.BuildMojo.execute (BuildMojo.java:211)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
[ERROR] F8: Build for fabric8-maven-sample-spring-boot-s2i failed: executor rejected
[ERROR] F8: fabric8-maven-sample-spring-boot-s2i-10		null	null
[ERROR] F8: Failed to execute the build: io.fabric8.maven.core.service.Fabric8ServiceException: Unable to build the image using the OpenShift build service
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 7.162 s
[INFO] Finished at: 2019-09-11T19:12:01+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.fabric8:fabric8-maven-plugin:4.3-SNAPSHOT:build (default-cli) on project fabric8-maven-sample-spring-boot: Failed to execute the build: io.fabric8.maven.core.service.Fabric8ServiceException: Unable to build the image using the OpenShift build service -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

```

* f-m-p version : 4.3-SNAPSHOT
* Maven version (`mvn -v`) :
```
Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-18T00:03:14+05:30)
Maven home: /usr/local/apache-maven
Java version: 1.8.0_212, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.212.b04-0.fc28.x86_64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.0.16-100.fc28.x86_64", arch: "amd64", family: "unix"

```
* Kubernetes / OpenShift setup and version : 
```
oc v3.11.0+0cbc58b
kubernetes v1.11.0+d4cacc0
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://192.168.42.51:8443
kubernetes v1.11.0+d4cacc0

```
* If it's a bug, how to reproduce :
Try building SNAPSHOT version of fmp, and build any sample on openshift.

**Comment by rohanKanojia** [comment]
when I revert back to kubernetes client 4.3.1, everything seems to be working fine.

**Comment by rohanKanojia** [comment]
This issue has been fixed in 4.5.2
